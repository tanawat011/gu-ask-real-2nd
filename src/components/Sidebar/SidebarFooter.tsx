import { useTranslation } from 'react-i18next'
import tw from 'twin.macro'

import packages from '../../../package.json'

const TwContainer = tw.div`flex h-sidebar-footer flex-col items-center justify-center border-t py-2 text-sm leading-6 border-[#374151]`

type SidebarFooterProps = {
  fullSidebar: boolean
}

export const SidebarFooter: React.FC<SidebarFooterProps> = ({ fullSidebar }) => {
  const { t } = useTranslation('sidebar')

  return (
    <TwContainer>
      {fullSidebar && <span>{t('footer.powered-by')}</span>}
      <span>{packages.version}</span>
    </TwContainer>
  )
}
