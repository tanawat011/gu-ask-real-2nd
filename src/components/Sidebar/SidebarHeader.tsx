import { useTranslation } from 'react-i18next'
import tw, { styled } from 'twin.macro'

type SidebarHeaderProps = {
  fullSidebar: boolean
}

const TwContainer = styled.div(() => [tw`flex h-sidebar-header items-center justify-between px-6`])
const TwWrapLogo = tw.div`flex h-full items-center`
const TwLogo = tw.div`h-8 w-8 rounded-full bg-cadet-grey`
const TwTitle = tw.div`ml-2 select-none text-white`

export const SidebarHeader: React.FC<SidebarHeaderProps> = ({ fullSidebar }) => {
  const { t } = useTranslation('sidebar')

  return (
    <TwContainer>
      <TwWrapLogo>
        <TwLogo />

        {fullSidebar && <TwTitle>{t('header.title')}</TwTitle>}
      </TwWrapLogo>
    </TwContainer>
  )
}
