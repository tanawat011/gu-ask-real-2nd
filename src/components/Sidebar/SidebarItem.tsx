import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import tw, { styled } from 'twin.macro'

export type SidebarItemProps = {
  icon?: JSX.Element
  label?: string
  hasChildren?: boolean
  onClick?: () => void
  selected?: boolean
  fullSidebar: boolean
}

type SidebarItemContainerProps = Pick<SidebarItemProps, 'selected' | 'fullSidebar'>

const TwContainer = styled.div(({ selected, fullSidebar }: SidebarItemContainerProps) => [
  tw`mb-2 flex h-10 cursor-pointer items-center rounded-lg hover:(text-white bg-[hsla(0,0%,100%,.1)])`,
  selected && tw`text-white bg-[hsla(0,0%,100%,.1)]`,
  fullSidebar ? tw`justify-between px-3` : tw`justify-center px-0`,
])
const TwItem = tw.div`flex items-center`
const TwWrapIcon = styled.div(({ fullSidebar }: SidebarItemContainerProps) => [
  tw`flex h-6 w-6 items-center justify-center`,
  fullSidebar && tw`mr-2`,
])

export const SidebarItem: React.FC<SidebarItemProps> = ({
  icon,
  label,
  hasChildren,
  onClick,
  selected,
  fullSidebar,
}) => {
  return (
    <TwContainer onClick={onClick} selected={selected} fullSidebar={fullSidebar}>
      <TwItem>
        <TwWrapIcon fullSidebar={fullSidebar}>{icon}</TwWrapIcon>

        {fullSidebar && <span className='mt-0.5'>{label}</span>}
      </TwItem>

      {fullSidebar && hasChildren && <FontAwesomeIcon icon='angle-down' size='lg' />}
    </TwContainer>
  )
}
