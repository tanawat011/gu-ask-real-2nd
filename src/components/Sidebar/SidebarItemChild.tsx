import type { SidebarState } from 'recoils/atoms'

import tw, { styled } from 'twin.macro'

export type SidebarItemProps = {
  label: string
  onClick?: () => void
  selected?: boolean
  fullSidebar?: boolean
}

type SidebarItemContainerProps = Pick<SidebarItemProps, 'selected'> & SidebarState

const TwContainer = styled.div(({ selected, fullSidebar }: SidebarItemContainerProps) => [
  tw`flex cursor-pointer items-center justify-between rounded-lg`,
  selected && tw`text-white bg-[hsla(0,0%,100%,.1)]`,
  fullSidebar
    ? tw`ml-5 h-10 pl-3 hover:(text-white bg-[hsla(0,0%,100%,.1)])`
    : tw`ml-0 h-9 px-4 py-1 text-[#9ca3af] hover:text-white`,
])
const TwItem = tw.div`flex items-center`

export const SidebarItemChild: React.FC<SidebarItemProps> = ({
  label,
  onClick,
  selected,
  fullSidebar,
}) => {
  return (
    <TwContainer onClick={onClick} selected={selected} fullSidebar={fullSidebar || false}>
      <TwItem>
        <span className='mt-0.5'>{label}</span>
      </TwItem>
    </TwContainer>
  )
}
