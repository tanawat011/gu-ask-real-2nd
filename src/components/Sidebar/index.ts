export * from './Sidebar'
export * from './SidebarHeader'
export * from './SidebarFooter'
