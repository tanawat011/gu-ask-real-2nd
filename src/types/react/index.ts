export * from './children'
export * from './reactRouterDom'
