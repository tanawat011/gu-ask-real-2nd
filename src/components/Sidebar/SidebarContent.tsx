import tw, { styled } from 'twin.macro'

import { FlexCol } from 'components/Common'

export type MenuItem = {
  label: string
  to: string
}

export type SidebarContentProps = {
  menuList: MenuItem[]
  itemSelected: string
  handleClickItem: (key: string, item: MenuItem) => void
}

type TwSidebarContentProps = Partial<SidebarContentProps> & {
  isActive?: boolean
}

const Container = tw(FlexCol)`h-full w-full bg-space-blue pt-[26px]`
const Item = styled.div(({ isActive }: TwSidebarContentProps) => [
  tw`relative flex cursor-pointer flex-row items-center h-[40px] mb-[12px]`,
  isActive && tw`bg-iron-gray`,
])
const ItemActive = tw.div`absolute z-0 h-full rounded-r bg-toy-red w-[6px]`
const ItemLabel = styled.div(({ isActive }: TwSidebarContentProps) => [
  tw`font-thaisans-nue z-0 font-semibold text-iron-gray text-[18px] ml-[22px] pb-[6px]`,
  isActive && tw`text-white`,
])

export const SidebarContent: React.FC<SidebarContentProps> = ({
  menuList,
  itemSelected,
  handleClickItem,
}) => {
  return (
    <Container>
      {menuList.map((item, index) => {
        const keyName = item.to.split('/')[1]
        const isActive = keyName === itemSelected

        return (
          <Item
            key={`sidebar-menu-item-${index}`}
            onClick={() => handleClickItem(keyName, item)}
            isActive={isActive}
          >
            {isActive && <ItemActive />}

            <ItemLabel isActive={isActive}>{item.label}</ItemLabel>
          </Item>
        )
      })}
    </Container>
  )
}
