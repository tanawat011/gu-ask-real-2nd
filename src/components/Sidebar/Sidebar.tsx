import tw, { styled } from 'twin.macro'

import { SidebarContent } from './SidebarContent'
import { SidebarFooter } from './SidebarFooter'
import { SidebarHeader } from './SidebarHeader'

export type MenuItemChildren = {
  to?: string
  label: string
  icon?: JSX.Element
  children?: Pick<MenuItemChildren, 'to' | 'label' | 'selected'>[]
  expanded?: boolean
  selected?: boolean
}

export type MenuItem = {
  title: string
  children: MenuItemChildren[]
}

export type SidebarProps = {
  menuList: MenuItem[]
  setMenuList: (menuList: MenuItem[]) => void
  fullSidebar: boolean
}

export type TwSidebarProps = Pick<SidebarProps, 'fullSidebar'>

const TwSidebar = styled.div(({ fullSidebar }: TwSidebarProps) => [
  tw`border-r min-w-[theme(width.sidebar)] border-[#374151]`,
  fullSidebar ? tw`min-w-[theme(width.sidebar)]` : tw`min-w-[theme(width.mini-sidebar)]`,
])

export const Sidebar: React.FC<SidebarProps> = ({ menuList, setMenuList, fullSidebar }) => {
  return (
    <TwSidebar fullSidebar={fullSidebar}>
      <SidebarHeader fullSidebar={fullSidebar} />

      <SidebarContent fullSidebar={fullSidebar} menuList={menuList} setMenuList={setMenuList} />

      <SidebarFooter fullSidebar={fullSidebar} />
    </TwSidebar>
  )
}
