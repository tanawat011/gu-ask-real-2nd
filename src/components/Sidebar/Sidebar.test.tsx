import { fireEvent, screen } from '@testing-library/react'

import { renderWithProviders } from 'utils/jest'

import { Sidebar } from './Sidebar'

describe('<Sidebar />', () => {
  test('should render without crashing and match snapshot', () => {
    expect.hasAssertions()

    const menuList = [
      {
        title: 'APPS',
        children: [
          {
            to: '/',
            label: 'Todo',
            icon: <div>xxx</div>,
          },
          {
            to: '/',
            label: 'Article',
            icon: <div>xxx</div>,
          },
        ],
      },
    ]

    const { asFragment } = renderWithProviders(
      <Sidebar menuList={menuList} setMenuList={jest.fn()} />,
    )

    const menuItemButton1 = screen.getByText(/menu-2/i)
    fireEvent.click(menuItemButton1)

    const menuItemButton2 = screen.getByText(/menu-1/i)
    fireEvent.click(menuItemButton2)

    const toggleSidebarButton = screen.getByTestId('hamburger-btn')
    fireEvent.click(toggleSidebarButton)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render with mini sidebar', () => {
    expect.hasAssertions()

    const menuList = [
      {
        title: 'APPS',
        children: [
          {
            to: '/',
            label: 'Todo',
            icon: <div>xxx</div>,
          },
          {
            to: '/',
            label: 'Article',
            icon: <div>xxx</div>,
          },
        ],
      },
    ]

    const { asFragment } = renderWithProviders(
      <Sidebar menuList={menuList} setMenuList={jest.fn()} />,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
