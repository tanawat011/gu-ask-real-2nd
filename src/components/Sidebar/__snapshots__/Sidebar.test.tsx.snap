// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should render with mini sidebar 1`] = `
<DocumentFragment>
  .c10 {
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-radius: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
}

.c10:hover {
  background-color: hsla(0,0%,100%,.1);
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  margin-right: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  width: 1.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  height: calc(100% - 64px - 64px - 1px);
}

.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-bottom: 1rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 600;
  overflow-y: overlay;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-top-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgb(55 65 81 / var(--tw-border-opacity));
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  height: 2rem;
  width: 2rem;
  border-radius: 9999px;
  --tw-bg-opacity: 1;
  background-color: rgb(156 163 175 / var(--tw-bg-opacity));
}

.c4 {
  margin-left: 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
}

.c5 {
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
}

.c5:hover {
  border-radius: 9999px;
  --tw-bg-opacity: 0.4;
  background-color: rgb(0 0 0 / var(--tw-bg-opacity));
}

.c0 {
  min-width: 290px;
  border-right-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgb(55 65 81 / var(--tw-border-opacity));
  min-width: 290px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
        <div
          class="c4"
        >
          header.title
        </div>
      </div>
      <div
        class="c5"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars fa-lg "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="c6 c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          APPS
        </div>
        <ul>
          <div
            class="c10"
          >
            <div
              class="c11"
            >
              <div
                class="c12"
              >
                <div>
                  xxx
                </div>
              </div>
              <span
                class="mt-0.5"
              >
                Todo
              </span>
            </div>
          </div>
          <div
            class="c10"
          >
            <div
              class="c11"
            >
              <div
                class="c12"
              >
                <div>
                  xxx
                </div>
              </div>
              <span
                class="mt-0.5"
              >
                Article
              </span>
            </div>
          </div>
        </ul>
      </div>
    </div>
    <div
      class="c13"
    >
      <span>
        footer.powered-by
      </span>
      <span>
        0.1.0
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Sidebar /> should render without crashing and match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c13 {
  position: relative;
  min-height: 52px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  --tw-bg-opacity: 1;
  background-color: rgb(24 28 35 / var(--tw-bg-opacity));
  padding-bottom: 10px;
}

.c14 {
  font-family: ThaiSans Neue;
  font-size: 18px;
  line-height: 18px;
  --tw-text-opacity: 1;
  color: rgb(99 105 118 / var(--tw-text-opacity));
}

.c14:first-child {
  padding-top: 14px;
}

.c3 {
  position: relative;
  min-height: 78px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  --tw-bg-opacity: 1;
  background-color: rgb(24 28 35 / var(--tw-bg-opacity));
  padding-left: 22px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  margin-left: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
}

.c6 {
  right: 6px;
  margin-top: 4px;
  margin-right: 14px;
  height: 22px;
  width: 22px;
  cursor: pointer;
}

.c1 {
  position: absolute;
  height: 100%;
  min-width: 220px;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  left: 0px;
  --tw-translate-x: 0px;
  -webkit-transform: translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  -ms-transform: translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  transform: translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
}

.c7 {
  height: 100%;
  width: 100%;
  --tw-bg-opacity: 1;
  background-color: rgb(24 28 35 / var(--tw-bg-opacity));
  padding-top: 26px;
}

.c8 {
  position: relative;
  margin-bottom: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-bg-opacity: 1;
  background-color: rgb(28 35 44 / var(--tw-bg-opacity));
}

.c11 {
  position: relative;
  margin-bottom: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  position: absolute;
  z-index: 0;
  height: 100%;
  width: 6px;
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
  --tw-bg-opacity: 1;
  background-color: rgb(231 52 57 / var(--tw-bg-opacity));
}

.c10 {
  font-family: ThaiSans Neue;
  z-index: 0;
  margin-left: 22px;
  padding-bottom: 6px;
  font-size: 18px;
  font-weight: 600;
  --tw-text-opacity: 1;
  color: rgb(99 105 118 / var(--tw-text-opacity));
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
}

.c12 {
  font-family: ThaiSans Neue;
  z-index: 0;
  margin-left: 22px;
  padding-bottom: 6px;
  font-size: 18px;
  font-weight: 600;
  --tw-text-opacity: 1;
  color: rgb(99 105 118 / var(--tw-text-opacity));
}

<div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    >
      <div
        class="c2 c4"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-house "
          color="#fff"
          data-icon="house"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
            fill="currentColor"
          />
        </svg>
        <div
          class="c5"
        >
          hello
        </div>
      </div>
      <div
        class="c6"
        data-testid="hamburger-btn"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars "
          color="#fff"
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="c0 c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        />
        <div
          class="c10"
        >
          menu-1
        </div>
      </div>
      <div
        class="c11"
      >
        <div
          class="c12"
        >
          menu-2
        </div>
      </div>
    </div>
    <div
      class="c0 c13"
    >
      <span
        class="c14"
      >
        hello
      </span>
      <span
        class="c14"
      >
        0.1.0
      </span>
    </div>
  </div>
</DocumentFragment>
`;
